turbine/src/broadcast_stage/standard_broadcast_run.rs

```diff
@@ -242,29 +243,30 @@ impl StandardBroadcastRun {
                 // Refrain from generating shreds for the slot.
                 return Err(Error::DuplicateSlotBroadcast(bank.slot()));
             }
+
             // Reinitialize state for this slot.
-            let chained_merkle_root = (self.slot == bank.parent_slot())
-                .then_some(self.chained_merkle_root)
-                .ok_or_else(|| {
-                    broadcast_utils::get_chained_merkle_root_from_parent(
-                        bank.slot(),
-                        bank.parent_slot(),
-                        blockstore,
-                    )
-                })
+
+            if self.slot != bank.parent_slot() {
```

Comment:
As mentioned on slack, I appreciate you looking into this code, and it is always good to have more eyes on the code and try to spot bugs or inefficiencies.

However this is only a "style" change and people's personal preference on what style is better can be different.
I personally prefer the old code because it avoids this `if` branch which changes state, whereas the old code uses an "expression". This type of state changing `if` branches tend to be a source of bugs and errors and I personally rather avoid them when possible.